src/main/java/com/atlassian/jira/cloud/jenkins/common/client/JenkinsAppApi.java

```diff
@@ -38,35 +44,55 @@ protected ResponseEntity sendRequest(
             throws ApiUpdateFailedException {
         try {
             final String requestPayload = objectMapper.writeValueAsString(jenkinsAppRequest);
-            final Request request = createRequest(webhookUrl, requestPayload);
+            RequestBody body = RequestBody.create(JSON_CONTENT_TYPE, requestPayload);
+            Request request = new Request.Builder().url(webhookUrl).post(body).build();
             final Response response = httpClient.newCall(request).execute();
-
             checkForErrorResponse(response);
+            return handleResponseBody(response, responseClass);
+        } catch (Exception e) {
+            throw handleError(e);
+        }
+    }
 
+    protected ResponseEntity sendRequestAsJwt(
+            final String webhookUrl,
+            final String secret,
+            final JenkinsAppRequest jenkinsAppRequest,
+            final Class<ResponseEntity> responseClass)
+            throws ApiUpdateFailedException {
+        try {
+            final String requestPayload = wrapInJwt(jenkinsAppRequest, secret);
+            RequestBody body = RequestBody.create(JWT_CONTENT_TYPE, requestPayload);
```

Comment:
This is the interesting part. The request will be of content type `application/jwt` and contain nothing but a JWT in the body.